pkg/builders/s2i/assemblers.go

```diff
@@ -0,0 +1,63 @@
+package s2i
+
+import (
+	"fmt"
+
+	fn "knative.dev/func/pkg/functions"
+)
+
+// GoAssembler
+//
+// Adapted from /usr/libexec/s2i/assemble within the UBI-8 go-toolchain
+// such that the "go build" command builds subdirectory .s2i/builds/by-func/last
+// (where main resides) rather than the root.
+// TODO: many apps use the pattern of having main in a subdirectory, for
+// example the idiomatic "./cmd/myapp/main.go".  It would therefore be
+// beneficial to submit a patch to the go-toolchain source allowing this
+// path to be customized with an environment variable instead
+const GoAssembler = `
+#!/bin/bash
+set -e
+pushd /tmp/src
+if [[ $(go list -f {{.Incomplete}}) == "true" ]]; then
+    INSTALL_URL=${INSTALL_URL:-$IMPORT_URL}
```

Comment:
`IMPORT_URL` is from where?

Reply:
This is just a copy-paste from the script which exists in the image already.  We very likely do not need these features ourselves, but I did not want to remove any, should other users expect them.